Plugin/CppApi/source/AbstractTool.cpp

```diff
@@ -18,39 +18,103 @@ namespace ArcGISRuntime
 {
 namespace Toolkit
 {
+/*!
+  \class Esri::ArcGISRuntime::Toolkit::AbstractTool
+  \inmodule ArcGISQtToolkit
+  \ingroup ToolCoordinateConversion
+  \ingroup ToolArcGISCompass
+  \brief Base class for classes that implement individual tools.
+  \since Esri::ArcGISRuntime 100.2
 
+  AbstractTool provides functions and properties that all derived tools share.
+ */
+
+/*!
+   \brief The constructor that accepts an optional \a parent object.
+ */
 AbstractTool::AbstractTool(QObject* parent /*= nullptr*/):
   QObject(parent)
 {
 
 }
 
+/*!
+   \brief The destructor.
+ */
 AbstractTool::~AbstractTool()
 {
 
 }
 
+/*!
+  \fn virtual bool AbstractTool::handleClick(const Point& pos)
+  \brief Reimplement this method in subclasses to handle a click at screen position \a pos.
+
+  Returns \c false.
+ */
 bool AbstractTool::handleClick(const Point& pos)
 {
   Q_UNUSED(pos)
   return false;
 }
 
+/*!
+  \fn virtual void AbstractTool::setProperties(const QVariantMap& properties)
+  \brief Reimplement this method in subclasses to set tool properties.
+  \list
+  \li \a properties - A QVariantMap containing property settings.
+  \endlist
```

Comment:
maybe we should say something along the lines of:

"A tool property is defined as a key value pairing where the key is a QString and the value can be any QVariant type. This can be useful for general settings which you wish to persist for the tool - for example default modes."